custom_components/tahoma/config_flow.py

```diff
@@ -112,56 +124,54 @@ async def async_step_user(
             errors=errors,
         )
 
-    async def async_step_reauth(
-        self, user_input: dict[str, Any] | None = None
-    ) -> FlowResult:
-        """Perform reauth if the user credentials have changed."""
-        self._reauth_entry = self.hass.config_entries.async_get_entry(
-            self.context["entry_id"]
-        )
-        self._default_username = user_input[CONF_USERNAME]
-        self._default_hub = user_input[CONF_HUB]
-
-        return await self.async_step_user()
-
-    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo):
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
         """Handle DHCP discovery."""
         hostname = discovery_info.hostname
         gateway_id = hostname[8:22]
 
-        _LOGGER.debug("DHCP discovery detected gateway %s", obfuscate_id(gateway_id))
-
-        if self._gateway_already_configured(gateway_id):
-            _LOGGER.debug("Gateway %s is already configured", obfuscate_id(gateway_id))
-            return self.async_abort(reason="already_configured")
-
-        return await self.async_step_user()
+        _LOGGER.debug(
+            "DHCP discovery detected gateway %s on host %s",
+            obfuscate_id(gateway_id),
+            discovery_info.hostname,
```

Comment:
Why this change? (and do we need to make this in core as well?)

Reply:
But doesn't this print gateway-xxxxx? (so the raw information we use for gateway id?)

Reply:
hmm possible! I was trying to print the ip address.

Reply:
Hostname is not the ip address in this case. I think best would to remove both of these and just keep it the same as in core.

Reply:
goes with 
```py
        _LOGGER.debug(
            "ZeroConf discovery detected gateway %s on host %s",
            obfuscate_id(gateway_id),
            discovery_info.host,
        )
```